File: work/src/MIL-32/Examples/General/MdigGrab/C#/MdigGrab.cs
//********************************************************************************
//
// File name: MdigGrab.cs
//
// Synopsis:  This program demonstrates how to grab from a camera in
//            continuous and monoshot mode.
//********************************************************************************
using System;
using System.Collections.Generic;
using System.Text;

using Matrox.MatroxImagingLibrary;

namespace MDigGrab
{
    class Program
    {
        static void Main(string[] args)
        {
            MIL_ID MilApplication = MIL.M_NULL;         // Application identifier.
            MIL_ID MilSystem = MIL.M_NULL;              // System identifier.
            MIL_ID MilDisplay = MIL.M_NULL;             // Display identifier.
            MIL_ID MilDigitizer = MIL.M_NULL;           // Digitizer identifier.
            MIL_ID MilImage = MIL.M_NULL;               // Image buffer identifier.

            // Allocate defaults.
            MIL.MappAllocDefault(MIL.M_DEFAULT, ref MilApplication, ref MilSystem, ref MilDisplay, ref MilDigitizer, ref MilImage);

            // Grab continuously.
            MIL.MdigGrabContinuous(MilDigitizer, MilImage);

            // When a key is pressed, halt.
            Console.Write("\nDIGITIZER ACQUISITION:\n");
            Console.Write("----------------------\n\n");
            Console.Write("Continuous image grab in progress.\n");
            Console.Write("Press <Enter> to stop.\n\n");
            Console.ReadKey();

            // Stop continuous grab.
            MIL.MdigHalt(MilDigitizer);

            // Pause to show the result.
            Console.Write("Continuous grab stopped.\n\n");
            Console.Write("Press <Enter> to do a single image grab.\n\n");
            Console.ReadKey();

            // Monoshot grab.
            MIL.MdigGrab(MilDigitizer, MilImage);

            // Pause to show the result.
            Console.Write("Displaying the grabbed image.\n");
            Console.Write("Press <Enter> to end.\n\n");
            Console.ReadKey();

            // Free defaults.
            MIL.MappFreeDefault(MilApplication, MilSystem, MilDisplay, MilDigitizer, MilImage);
        }
    }
}

File: work/src/MIL-32/Examples/General/MdigGrab/VB/MdigGrab.vb
'********************************************************************************
'
' File name: MdigGrab.vb
'
' Synopsis:  This program demonstrates how to grab from a camera in
'            continuous and monoshot mode.
'********************************************************************************
Imports Microsoft.VisualBasic
Imports System
Imports System.Collections.Generic
Imports System.Text

Imports Matrox.MatroxImagingLibrary

Namespace MDigGrab
    Friend Class Program
        Shared Sub Main(ByVal args() As String)
            Dim MilApplication As MIL_ID = MIL.M_NULL ' Application identifier.
            Dim MilSystem As MIL_ID = MIL.M_NULL ' System identifier.
            Dim MilDisplay As MIL_ID = MIL.M_NULL ' Display identifier.
            Dim MilDigitizer As MIL_ID = MIL.M_NULL ' Digitizer identifier.
            Dim MilImage As MIL_ID = MIL.M_NULL ' Image buffer identifier.

            ' Allocate defaults.
            MIL.MappAllocDefault(MIL.M_DEFAULT, MilApplication, MilSystem, MilDisplay, MilDigitizer, MilImage)

            ' Grab continuously.
            MIL.MdigGrabContinuous(MilDigitizer, MilImage)

            ' When a key is pressed, halt.
            Console.Write(Constants.vbLf + "DIGITIZER ACQUISITION:" + Constants.vbLf)
            Console.Write("----------------------" + Constants.vbLf + Constants.vbLf)
            Console.Write("Continuous image grab in progress." + Constants.vbLf)
            Console.Write("Press <Enter> to stop." + Constants.vbLf + Constants.vbLf)
            Console.ReadKey()

            ' Stop continuous grab.
            MIL.MdigHalt(MilDigitizer)

            ' Pause to show the result.
            Console.Write("Continuous grab stopped." + Constants.vbLf + Constants.vbLf)
            Console.Write("Press <Enter> to do a single image grab." + Constants.vbLf + Constants.vbLf)
            Console.ReadKey()

            ' Monoshot grab.
            MIL.MdigGrab(MilDigitizer, MilImage)

            ' Pause to show the result.
            Console.Write("Displaying the grabbed image." + Constants.vbLf)
            Console.Write("Press <Enter> to end." + Constants.vbLf + Constants.vbLf)
            Console.ReadKey()

            ' Free defaults.
            MIL.MappFreeDefault(MilApplication, MilSystem, MilDisplay, MilDigitizer, MilImage)
        End Sub
    End Class
End Namespace
